src/lib.rs

```diff
@@ -208,6 +208,21 @@ pub trait Integer: Sized + Num + PartialOrd + Ord + Eq {
         (self.extended_gcd(other), self.lcm(other))
     }
 
+    /// Inverse of the number modulo the modulus
+    #[inline]
+    fn invmod(&self, modulus: &Self) -> Self
+    where
+        Self: Clone,
+    {
+        let egcd = self.extended_gcd(modulus);
+        assert!(egcd.gcd.is_one(), "no inverse, GCD(x, modulus) != 1");
```

Comment:
I think we should return `Option<Self>`, with `None` here instead of panicking.

Reply:
For sure, wasn't sure what was preferred. Will change return to `Option<Self>`.